src/Data/Range.hs

```diff
@@ -42,9 +32,6 @@ subtractRange range1 range2 = Range (start range1) (end range1 - rangeLength (Ra
 instance Semigroup Range where
   Range start1 end1 <> Range start2 end2 = Range (min start1 start2) (max end1 end2)
 
-instance Ord Range where
-  a <= b = start a <= start b
```

Comment:
As far as I can recall, I defined this in the first place because I didn’t understand either a) the rules that derived `Ord` instances obey (especially w.r.t. lexicographical ordering), or b) the laws governing the expected behaviour of `Ord` instances in general.

One such law is that `Ord` instances should agree with `Eq` instances (`Eq` is a superclass of `Ord`, after all), which this instance violated: `x <= y && y <= x = x == y` doesn’t hold; equivalently, `compare x y == EQ = x == y` doesn’t hold.